dev/archery/archery/crossbow/core.py

```diff
@@ -537,17 +537,36 @@ def _latest_prefix_id(self, prefix):
             latest = -1
         return latest
 
+    def _latest_prefix_date(self, prefix):
+        pattern = re.compile(r'[\w\/-]*{}-(\d+)-(\d+)-(\d+)'.format(prefix))
+        matches = list(filter(None, map(pattern.match, self.repo.branches)))
+        if matches:
+            latest = sorted([m.group(0) for m in matches])[-1]
+            # slice the trailing date part (YYYY-MM-DD)
+            latest = latest[-10:]
+        else:
+            latest = -1
+        return latest
```

Comment:
Stylistic suggestion:
```python
# slice [-10:] corresponds to the trailing date part (YYYY-MM-DD)
return sorted([m.group(0) for m in matches])[-1][-10:] if matches else -1
```

Reply:
That's just too long to fit on a single line unfortunately, so then it might not actually get that much more readable if the one-liner uses line continuation .. So will leave as is.